### ISSUE TITLE
Multiline method definitions

### ISSUE THREAD

**Issue by fny** [rubocop-hq/ruby-style-guide#241]
Given the following mess:

``` ruby
def some_unfortunate_method(with, an, inordinately, long, list, of, required, arguments, that, also, has, options = {})
end
```

This seems most appropriate:

``` ruby
def some_unfortunate_method(with, an, inordinately, long, list, of, required,
                            arguments, that, also, has, options = {})
end
```

**Comment by yemartin** [comment]
@bbatsov I just hit an issue with a long method definition, even though it only had few arguments. The problem can easily arise when these arguments also have default values. So having a rule about avoiding methods with lots of arguments does not guarantee not having to deal with long method definitions. 

So please, could you reopen this?


For example, this has only 3 arguments, but is pretty aweful:

```ruby
def some_unfortunate_method(some_param, another_param: SomeModule.some_method("a.quite.looooooooooooooong.string.to.look.up"), third_param: SomeModule.some_method("another.looooooooooooooong.string.to.look.up"))
  my
  method
  body
end
```

Inspired by the "good (normal indent)" format in [method call parameter indentation guideline](https://github.com/rubocop-hq/ruby-style-guide#no-double-indent), I did this:

```ruby
def some_unfortunate_method(
  some_param,
  another_param: SomeModule.some_method("a.quite.looooooooooooooong.string.to.look.up"),
  third_param: SomeModule.some_method("another.looooooooooooooong.string.to.look.up")
)
  my
  method
  body
end
```


### Bottom Line

It looks to me like the [method *call* parameter indentation guideline](https://github.com/rubocop-hq/ruby-style-guide#no-double-indent) can apply just equally well to method *definitions*. So the fix to this issue could simply be a small two words addition:

> Align the parameters of a method call **or definition** if they span more than one line.

What do you think?

**Comment by Eneroth3** [comment]
I think this is a valid question. A method doesn't need to have a lot of arguments to fill up a line if only the names are long, and I think variables with pronounceable real word names are to be preferred over abbreviations or ambiguous names, or if they have default values. While many cases probably can be refactored I every now and then end up in a scenario where the most readable solution requires more than a full line.

Consider this example:

```ruby
# SketchUp extensions all run in a shared environment on the end user's machine,
# so developers need to wrap them in a personal namespace to avoid collision.
module MyDeveloperName
  module SomeExtension
    module SomeModuleInsideMyExtension
      # Perform a boolean intersect operation between two component definitions.
      #
      # @param definition1 [Sketchup::ComponentDefinition]
      # @param definition2 [Sketchup::ComponentDefinition]
      # @param transformation1 [Geom::Transformation]
      # @param transformation2 [Geom::Transformation]
      #
      # @return [Object] Some return value.
      def intersect(definition1, definition2, transformation1 = IDENTITY, transformation2 = IDENTITY)
        #...
      end
    end
  end
end
```

I could make the arguments shorter by using other names, but a definition and transformation are well established names in this context. I could use an option hash but that would just make the calls to the method messier; two objects to intersect and their (optional) positions is quite self-explanatory and keys would just be clutter.

I could convert the arguments to an array using *args and use that, but then YARD wouldn't pick up the default values. I could create a new constant `I`, referring to `IDENTITY`, the constant the SketchUp Ruby API defines for the identity transformation matrix, but no one would understand what that means without reading elsewhere.

All workarounds I can come up with are really just that: workarounds. They don't help increase readability, only to fulfill rules. I think a rule for what to do with multi-line arguments would be useful in these cases.
